src/services/TableService.js

```diff
@@ -149,53 +192,15 @@ function TableService($q, $http, URLService, UtilsService, CacheService, _, Elem
               readTablesCommon();
             });
         }
-        else {
-          var tableContains = [];
-          for ( k = 0; k < data.contains.length; k++ ){
-            if ( data.contains[k].type ==="Table"){
-              //use table title or text before the table as tableTitles
-              if ( data.contains[k].title !== undefined &&  data.contains[k].title.length !== 0)
-                tableTitles.push(toValidId(data.contains[k].title)); 
-              else if ( data.contains[k-1].sourceType==="text")
-                tableTitles.push(toValidId(data.contains[k-1].text.replace("<p>","").replace("</p>","").replace(" ", ""))); //assume it is Paragraph
-              else
-                tableTitles.push("");
-
-              tableIds.push("_" + Math.floor((Math.random() * 1000) + 1)); //a random between 1 and 1000
-              tableContains.push(data.contains[k]);
-              var columnHeaders = [];
-              //assume first column is empty
-              for ( var kk = 1; kk < data.contains[k].header[0].length; kk++){
-                columnHeaders[kk-1] = data.contains[k].header[0][kk].content[0].text.replace("<p>","").replace("</p>","");
-              }  
-              tableColumnHeadersLabels.push(columnHeaders); //xxx, yyy, mass,cost, power in string
-            }
-          }
-          var body;
-          
-          for ( k = 0; k < tableContains.length; k++){
-              body = tableContains[k].body;
-              numOfRowHeadersPerTable.push( body.length);
-              for (i = 0; i < body.length; i++ ){
-                rowHeadersMmsEid.push(body[i][0].content[0].source);
-                for ( j = 1; j < body[i].length; j++){
-                  dataValuesMmmEid.push(body[i][j].content[0].source);
-                if ( i === 0)
-                  columnCounter = body[i].length - 1;// -1 to remove row header
-              }
-              numOfDataColumn.push(columnCounter);
-            }
-          } //end of for k
-          readTablesCommon();
-        }
-
+ 
         //requires numOfRowHeadersPerTable, rowHeadersMmsEid, dataValuesMmmEid, tableIds(to get number of tables)
         //common function to read 2.2 and previous version JSON
         function readTablesCommon(){
           var numOfTables = tableIds.length;
-          ElementService.getElements(rowHeadersMmsEid, false, ws, version)
+
+          ElementService.getElements(rowHeadersMmsEid, 1, false)
           .then(function(rowHeaders) {
-              ElementService.getElements(dataValuesMmmEid, false, ws, version)
+              ElementService.getElements(dataValuesMmmEid, 1, false)
```

Comment:
the results returned from ElementService.getElements are not necessarily in the same order, may need to account for that